Assets/HoloToolkit/Interaction/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,264 @@
+        }
+
+		/// <summary>
+		/// Register an interactible with this receiver.
+		/// </summary>
+		/// <param name="interactible">takes a GameObject as the interactible to register.</param>
+		public virtual void RegisterInteractible(GameObject interactible)
+		{
+			if (interactible == null || Interactibles.Contains(interactible))
+				return;
+
+			Interactibles.Add(interactible);
+		}
+
+#if UNITY_EDITOR
+		/// <summary>
+		/// When selected draw lines to all linked interactibles
+		/// </summary>
+		protected virtual void OnDrawGizmosSelected()
+		{
+			if (this.Interactibles.Count > 0)
+			{
+				GameObject[] bioList = this.Interactibles.ToArray();
+
+				for (int i = 0; i < bioList.Length; i++)
+				{
+					if (bioList[i] != null)
+					{
+						Gizmos.color = Color.green;
+						Gizmos.DrawLine(this.transform.position, bioList[i].transform.position);
+					}
+				}
+			}
+
+			if (this.Targets.Count > 0)
+			{
+				GameObject[] targetList = this.Targets.ToArray();
+
+				for (int i = 0; i < targetList.Length; i++)
+				{
+					if (targetList[i] != null)
+					{
+						Gizmos.color = Color.red;
+						Gizmos.DrawLine(this.transform.position, targetList[i].transform.position);
+					}
+				}
+			}
+		}
+
+		/// <summary>
+		/// On Draw Gizmo show the receiver icon
+		/// </summary>
+		protected virtual void OnDrawGizmos()
+		{
+			_gizmoIcon = UnityEditor.Selection.activeGameObject == this.gameObject ? _gizmoIconSelected : _gizmoIconDefault;
+			Gizmos.DrawIcon(this.transform.position, _gizmoIcon, false);
+		}
+#endif
+
+		/// <summary>
+		/// Function to remove an interactible from the linked list.
+		/// </summary>
+		/// <param name="interactible"></param>
+		public virtual void RemoveInteractible(GameObject interactible)
+		{
+			if (interactible != null && Interactibles.Contains(interactible))
+			{
+				Interactibles.Remove(interactible);
+			}
+		}
+
+		/// <summary>
+		/// Clear the interactibles list and unregister them
+		/// </summary>
+		public virtual void ClearInteractibles()
+		{
+			GameObject[] _intList = Interactibles.ToArray();
+
+			for (int i = 0; i < _intList.Length; i++)
+			{
+				this.RemoveInteractible(_intList[i]);
+			}
+		}
+
+		/// <summary>
+		/// Is the game object interactible in our list of interactibles
+		/// </summary>
+		/// <param name="interactible"></param>
+		/// <returns></returns>
+		protected bool IsInteractible(GameObject interactible)
+		{
+			return (Interactibles != null && Interactibles.Contains(interactible));
+		}
+
+		/// <summary>
+		/// Check if the object is in our list - if yes 
+		/// </summary>
+		/// <param name="sendEvent"></param>
+		/// <param name="obj"></param>
+		/// <param name="eventArgs"></param>
+		private void CheckAndSendEvent(System.Action<GameObject, UnityEngine.XR.WSA.Input.InteractionManager.SourceEventHandler> sendEvent, GameObject obj)
+		{
+			if (IsInteractible(obj))
+			{
+				// sendEvent(obj, eventArgs);
```

Comment:
What did this use to do?  Why was it removed?

Reply:
Sorry I had the wrong version in there.  That used to filter and send forward the events after filtering that it's coming from and interactible in the list.  Changed this to filter and only fire the protected virtual functions based on the interactible being in the list.  I'd like to clean this up on the InputManager side allowing for external subscriptions to specific objects in the future but wanted to not have any changes to core HTK manager scripts for now.

Should be resolved in the last commit.